src/main/java/net/minecraftforge/client/gui/config/widgets/ConfigGuiWidget.java

```diff
@@ -0,0 +1,740 @@
+        {
+            try
+            {
+                return isValid(parser.apply(value));
+            }
+            catch (NumberFormatException e)
+            {
+                return false;
+            }
+        }
+
+        private boolean isValid(T value)
+        {
+            return this.valueSpec.test(value);
+        }
+    }
+
+    /**
+     * Default implementation for a {@link ConfigGuiWidget} for a {@link ForgeConfigSpec.IntValue}.
+     */
+    public static class IntegerWidget extends NumberWidget<Integer>
+    {
+        public static ConfigGuiWidgetFactory FACTORY = (value, valueSpec, valueManager, spec, name) ->
+        {
+            if (value instanceof ForgeConfigSpec.IntValue)
+            {
+                return new IntegerWidget(valueSpec, valueManager, spec, name);
+            }
+
+            throw new IllegalArgumentException("The given config value for path: " + String.join(".",
+                    value.getPath()) + " is not a integer value!");
+        };
+
+        public IntegerWidget(final ForgeConfigSpec.ValueSpec valueSpec, final ValueManager valueManager,
+                             SpecificationData specificationData, final Component name)
+        {
+            super(valueSpec, valueManager, specificationData, name, Integer::parseInt);
+        }
+    }
+
+    /**
+     * Default implementation for a {@link ConfigGuiWidget} for a {@link ForgeConfigSpec.LongValue}.
+     */
+    public static class LongWidget extends NumberWidget<Long>
+    {
+        public static ConfigGuiWidgetFactory FACTORY = (value, valueSpec, valueManager, spec, name) ->
+        {
+            if (value instanceof ForgeConfigSpec.LongValue)
+            {
+                return new LongWidget(valueSpec, valueManager, spec, name);
+            }
+
+            throw new IllegalArgumentException("The given config value for path: " + String.join(".",
+                    value.getPath()) + " is not a long value!");
+        };
+
+        public LongWidget(final ForgeConfigSpec.ValueSpec valueSpec,
+                          final ValueManager valueManager, SpecificationData specificationData, final Component name)
+        {
+            super(valueSpec, valueManager, specificationData, name, Long::parseLong);
+        }
+    }
+
+    /**
+     * Default implementation for a {@link ConfigGuiWidget} for a {@link ForgeConfigSpec.DoubleValue}.
+     */
+    public static class DoubleWidget extends NumberWidget<Double>
+    {
+        public static ConfigGuiWidgetFactory FACTORY = (value, valueSpec, valueManager, spec, name) ->
+        {
+            if (value instanceof ForgeConfigSpec.DoubleValue)
+            {
+                return new DoubleWidget(valueSpec, valueManager, spec, name);
+            }
+
+            throw new IllegalArgumentException("The given config value for path: " + String.join(".",
+                    value.getPath()) + " is not a Double value!");
```

Comment:
Other number widgets you have the number in lower case (for example `long`). I am not sure which makes more sense (probably capital like this), but either way they should be consistent.